### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by denisdiaconu**
I created a Ruby on Rails application that allows the user to:

-    register and log in, so that the data is private to them.
-    introduce new transactions associated with a category.
-    see the money spent on each category.

### CODE REVIEW

File: app/views/entities/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      <h2><%= pluralize(entity.errors.count, "error") %> prohibited this entity from being saved:</h2>
+
+      <ul>
+        <% entity.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
```
Comment (Emmyn5600) [inline app/views/entities/_form.html.erb:13]: - [x]  We don't need to add any logic inside the .erb file, we use the helper folders to create methods with logic and called inside the .erb file. You can refer to this comment to modify the files that have logic inside the .erb file.

Read [this](https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make) for more information

File: app/views/groups/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(group.errors.count, "error") %> prohibited this group from being saved:</h2>
+
+      <ul>
+        <% group.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (Emmyn5600) [inline app/views/groups/_form.html.erb:12]: - [x]  We don't need to add any logic inside the .erb file, we use the helper folders to create methods with logic and called inside the .erb file. You can refer to this comment to modify the files that have logic inside the .erb file.

Read [this](https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make) for more information

File: app/views/shared/_msg.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<% if notice  %>
+<div class="alert alert-warning" role="alert">
+  <%= notice %>
+</div>
+<% end  %>
+<% if alert  %>
+<div class="alert alert-danger" role="alert">
+  <%= alert %>
+</div>
+<% end  %>
```
Comment (Emmyn5600) [inline app/views/shared/_msg.html.erb:10]: - [x]  We don't need to add any logic inside the .erb file, we use the helper folders to create methods with logic and called inside the .erb file. You can refer to this comment to modify the files that have logic inside the .erb file.

Read [this](https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make) for more information

File: test/controllers/entities_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  test 'should destroy entity' do
+    assert_difference('Entity.count', -1) do
+      delete entity_url(@entity)
+    end
+
+    assert_redirected_to entities_url
+  end
+end
```
Comment (Emmyn5600) [inline test/controllers/entities_controller_test.rb:52]: - [x] Optional suggetions 

You have used `minitest` for testing cases, however, you would have used `rspec`, kindly provide clear information in the readme file related to how to run the tests.

File: app/controllers/groups_controller.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  # GET /groups or /groups.json
+  def index
+    @groups = Group.all
+  end
+
+  # GET /groups/1 or /groups/1.json
+  def show
+    @group_entities = Entity.where(group_id: params[:id]).all
+  end
```
Comment (Emmyn5600) [inline app/controllers/groups_controller.rb:12]: - [x]  You should avoid making SQL queries outside of models. It makes it harder to test and debug your app.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi, @denisdiaconu 

Good job so far! 👏🏼 

- Your project is running successfully.
- You have completed almost all requirements.
- Database is well structured.
- No linter errors
- You have a descriptive Readme file

Great work on making the project requirements:clap:
You've done well implementing the project requirements, but there are still some issues which aren't been addressed yet 
 
## STATUS: CHANGES REQUESTED :recycle:

### Changes requested.

- [x] Check the comment under the review.

 #### Optional suggestions
 
 - N/A
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by descholar-ceo — APPROVED** [review]
Hi @denisdiaconu , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**